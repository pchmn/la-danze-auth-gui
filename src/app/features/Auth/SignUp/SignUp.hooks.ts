import { gql } from "@apollo/client/core";
import { useMutation } from "@apollo/client/react";
import { useFormik } from "formik";
import { GraphQLError } from "graphql/error/GraphQLError";
import { useTranslation } from "react-i18next";
import { useHistory, useRouteMatch } from "react-router-dom";
import * as yup from 'yup';
import { setStorage, StorageKey } from "../../../core/storage/appStorage";


// GraphQL
const SIGN_UP = gql`
  mutation Signup($input: SignUpInput!) {
    signUp(input: $input) {
      accessToken
    }
  }
`;

/**
 * Get SignUp hook
 * 
 * @returns  hook
 */
export function useSignUp() {
  // Translations
  const { t } = useTranslation();
  // Form
  const formik = createFormik();
  // GraphQL
  const [signUp, { data, loading, error }] = useMutation(SIGN_UP, {
    onError(err) {
      handleError(err.graphQLErrors)
    }
  });
  // Router
  const history = useHistory();
  const { path } = useRouteMatch();

  return { formik, data, loading, error, path, t };


  function handleError(errors: readonly GraphQLError[]) {
    errors.map((error: GraphQLError) => {
      if (error.extensions && error.extensions.exception && error.extensions.exception.code === 100) {
        formik.setErrors({ email: 'Email déjà utilisé' });
      }
    })
  }

  function handleSuccessSignup(accessToken: string) {
    // Save in localStorage
    setStorage(StorageKey.AccessToken, accessToken);
    // Go to my account
    history.push(`${path}/account-created`)
  }


  function createFormik() {
    const validationSchema = yup.object({
      email: yup
        .string()
        .email(t('common.errors.invalidField', { field: t('auth.signUp.email') }))
        .required(t('common.errors.requiredField')),
      username: yup
        .string()
        .min(2, t('common.errors.minLength', { min: 2 }))
        .required(t('common.errors.requiredField')),
      password: yup
        .string()
        .min(8, t('common.errors.minLength', { min: 8 }))
        .required(t('common.errors.requiredField')),
      confirmPassword: yup
        .string()
        .oneOf([yup.ref('password'), ''], t('auth.signUp.passwordsMismatch'))
        .required(t('common.errors.requiredField')),
    });

    return useFormik({
      initialValues: {
        email: '',
        username: '',
        password: '',
        confirmPassword: ''
      },
      validationSchema: validationSchema,
      onSubmit: (values) => {
        console.log({ email: values.email, username: values.username, password: values.password })
        signUp({ variables: { input: { email: values.email, username: values.username, password: values.password } } }).then(res => {
          console.log(res);
          handleSuccessSignup(res.data.accessToken);
        });
      },
    });
  }
}
