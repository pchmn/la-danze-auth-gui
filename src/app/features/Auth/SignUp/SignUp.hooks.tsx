import { gql } from "@apollo/client/core";
import { useMutation } from "@apollo/client/react";
import { yupResolver } from '@hookform/resolvers/yup';
import { GraphQLError } from "graphql/error/GraphQLError";
import { useState } from "react";
import { useForm, UseFormMethods } from "react-hook-form";
import { useTranslation } from "react-i18next";
import * as yup from 'yup';
import { setStorage, StorageKey } from "../../../core/storage/appStorage";


// GraphQL
const SIGN_UP = gql`
  mutation Signup($input: SignUpInput!) {
    signUp(input: $input) {
      accessToken
    }
  }
`;

/**
 * Get SignUp hook
 * 
 * @returns  hook
 */
export function useSignUp() {
  // Translations
  const { t } = useTranslation();
  // Form
  const form: UseFormMethods<any> & { onSubmit?: any } = useForm({
    resolver: yupResolver(createValidationSchema()),
  });
  form.onSubmit = onSubmit;
  // GraphQL
  const [signUp, { data, loading }] = useMutation(SIGN_UP, {
    onError(err) {
      handleError(err.graphQLErrors)
    }
  });
  // Local state
  const [signUpSuccess, setSignupSuccess] = useState(false)

  return { form, loading, t, signUpSuccess };

  function handleError(errors: readonly GraphQLError[]) {
    errors.map((error: GraphQLError) => {
      if (error.extensions && error.extensions.exception && error.extensions.exception.code === 100) {
        form.errors.email.type = 'alreadyUsed';
        form.errors.email.message = t('auth.signUp.emailAlreadyUsed');
      }
    })
  }

  function handleSuccessSignup(accessToken: string) {
    // Save in localStorage
    setStorage(StorageKey.AccessToken, accessToken);
  }

  function onSubmit(values: any) {
    console.log({ email: values.email, username: values.username, password: values.password })
    signUp({ variables: { input: { email: values.email, username: values.username, password: values.password } } })
      .then(res => {
        console.log(res);
        handleSuccessSignup(res.data.accessToken);
      })
      .catch(err => setSignupSuccess(true));
  }

  function createValidationSchema() {
    return yup.object({
      email: yup
        .string()
        .required(t('common.errors.requiredField'))
        .email(t('common.errors.invalidField', { field: t('auth.signUp.email') })),
      username: yup
        .string()
        .required(t('common.errors.requiredField'))
        .min(2, t('common.errors.minLength', { min: 2 })),
      password: yup
        .string()
        .required(t('common.errors.requiredField'))
        .min(8, t('common.errors.minLength', { min: 8 })),
      confirmPassword: yup
        .string()
        .required(t('common.errors.requiredField'))
        .oneOf([yup.ref('password'), ''], t('auth.signUp.passwordsMismatch')),
    });
  }
}
