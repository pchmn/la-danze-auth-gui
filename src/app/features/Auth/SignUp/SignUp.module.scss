.signInHelper {
  display: block;
  text-align: center;
  margin-top: 40px;
  margin-bottom: 10px;
  color: #979797;
  font-size: 0.9rem;

  @media (max-width: 530px) {
    margin-top: 20px;
  }

  a {
    font-weight: 500;
    font-size: 0.95rem;
    margin-left: 5px;
  }
}

.title {
  margin-top: 0;
  margin-bottom: 50px;
  font-size: 2rem;
  font-weight: 400;

  @media (max-width: 530px) {
    font-size: 1.8rem;
  }
}

.form {

  .passwordHelper {
    display: block;
    padding: 5px 14px 15px 14px;
    font-size: 0.9rem;
  }

  .textField {
    display: block;
    margin-bottom: 25px;
  }

  .submitButton {
    margin-top: 20px;
  }
}

.progress {
  position: absolute !important;
  bottom: 0px;
  left: 0;
  width: 100%;

  @media (max-width: 530px) {
    top: 0;
  }
}

.AccountCreated {
  text-align: center;
  display: flex;
  flex-direction: column;
  position: relative;
  flex: 1;

  .title {
    margin-bottom: 0;
  }

  @media (min-width: 530px) {
    .title {
      position: absolute;
      top: 0;
      left: 50%;
      transform: translateX(-50%);
      width: fit-content;
    }
  
    div {
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
  
      img {
        display: block;
        margin: auto;
        margin-bottom: 15px;
      }
    }
  
    button {
      position: absolute;
      bottom: 0;
      left: 50%;
      transform: translateX(-50%);
    }
  }

  @media (max-width: 530px) {
  
    div {
      text-align: center;
      margin-bottom: 100px;
      margin-top: 100px;
  
      img {
        display: block;
        margin: auto;
        margin-bottom: 15px;
      }
    }

    button {
      width: fit-content;
      margin-left: auto;
      margin-right: auto;
    }

  }
}