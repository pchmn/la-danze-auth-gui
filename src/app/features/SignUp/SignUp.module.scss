.SignUp {
  height: 100%;
  padding-top: 40px;

  @media (max-width: 530px) {
    padding-top: 20px;
  }

  .logo {
    display: block;
    margin: auto;
  }

  .paper {
    flex: 1;
    margin-top: 40px;
    padding: 70px;
    max-width: 530px;
    background: #3D3D3D;
    border-radius: 10px;
    
    @media (max-width: 530px) {
      padding: 50px;
      margin-top: 20px;
      height: fit-content;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    }
  
    .title {
      margin-top: 0;
      margin-bottom: 50px;
      font-size: 2rem;
      font-weight: 400;

      @media (max-width: 530px) {
        font-size: 1.8rem;
      }
    }
  
    .form {
      margin: auto;

      .passwordHelper {
        display: block;
        padding: 5px 14px 15px 14px;
        font-size: 0.9rem;
      }
  
      .textField {
        display: block;
        margin-bottom: 25px;
      }
  
      .submitButton {
        border-radius: 30px;
        padding: 6px 40px;
        margin-top: 10px;
      }
    }
  }
}

.leftPart {
  text-align: center;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;

  div {
    position: absolute;
    top: 30px;
    width: 100%;
  }


  h2 {
    font-size: 2.2rem;
  }
}

.rightPart {
  background-color: #3B3B3B;
  text-align: center;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
}