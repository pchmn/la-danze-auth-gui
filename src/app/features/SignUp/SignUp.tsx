import Button from '@material-ui/core/Button';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import TextField from '@material-ui/core/TextField';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { useFormik } from 'formik';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import * as yup from 'yup';
import styles from './SignUp.module.scss';


export function SignUp() {
  const dispatch = useDispatch();
  const matches = useMediaQuery('(max-width: 530px)');
  const { t } = useTranslation();

  const validationSchema = yup.object({
    email: yup
      .string()
      .email(t('common.errors.invalidField', { field: t('signUp.email') }))
      .required(t('common.errors.requiredField')),
    username: yup
      .string()
      .required(t('common.errors.requiredField')),
    password: yup
      .string()
      .min(8, t('common.errors.minLength', { min: 8 }))
      .required(t('common.errors.requiredField')),
    confirmPassword: yup
      .string()
      .oneOf([yup.ref('password'), ''], t('signUp.passwordsMismatch'))
      .required(t('common.errors.requiredField')),
  });

  const formik = useFormik({
    initialValues: {
      email: '',
      username: '',
      password: '',
      confirmPassword: ''
    },
    validationSchema: validationSchema,
    onSubmit: (values) => {
      console.log(values);
    },
  });

  function textField(field: 'email' | 'username' | 'password' | 'confirmPassword') {
    return <TextField
      className={styles.textField}
      fullWidth
      size="small"
      variant="outlined"
      id={field}
      name={field}
      type={['password', 'confirmPassord'].includes(field) ? 'password' : 'text'}
      label={t(`signUp.${field}`)}
      value={formik.values[field]}
      onChange={formik.handleChange}
      error={formik.touched[field] && Boolean(formik.errors[field])}
      helperText={formik.touched[field] && formik.errors[field]}
    />;
  }

  return (
    <Container className={`${styles.SignUp} ${matches ? 'flex-column' : ''}`} maxWidth="sm">
      <div><img className={styles.logo} alt="logo" src="./images/logo-48px.png" /></div>
      <Paper className={`${styles.paper} darker`}>
        <h2 className={styles.title}>{t('signUp.title')}</h2>
        <form className={styles.form} onSubmit={formik.handleSubmit}>
          {textField('email')}
          {textField('username')}
          <span className={styles.passwordHelper}>{t('signUp.passwordHelper')}</span>
          {textField('password')}
          {textField('confirmPassword')}
          <Button className={styles.submitButton} variant="contained" color="primary" type="submit">
            {t('signUp.validate')}
          </Button>
        </form>
      </Paper>
    </Container>
  );
}
