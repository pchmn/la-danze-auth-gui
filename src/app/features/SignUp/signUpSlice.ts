import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from '../../core/redux/store';

export interface SignUpState {
  activeStep: number;
  email: string;
  username: string;
  emailFormTouched: boolean;
}

const initialState: SignUpState = {
  activeStep: 0,
  email: '',
  username: '',
  emailFormTouched: false
};

export const signUpSlice = createSlice({
  name: 'signUp',
  initialState,
  reducers: {
    nextStep: state => {
      state.activeStep = (state.activeStep === 2 ? 2 : state.activeStep + 1);
    },
    previousStep: state => {
      state.activeStep = (state.activeStep === 0 ? 0 : state.activeStep - 1);
    },
    setEmail: (state, action: PayloadAction<string>) => {
      state.email = action.payload;
    },
    setUsername: (state, action: PayloadAction<string>) => {
      state.username = action.payload;
    },
    setFormTouched: state => {
      state.emailFormTouched = true;
    }
  },
});

export const { nextStep, previousStep, setEmail, setUsername, setFormTouched } = signUpSlice.actions;

export const selectActiveStep = (state: RootState) => state.signUp.activeStep;

export default signUpSlice.reducer;