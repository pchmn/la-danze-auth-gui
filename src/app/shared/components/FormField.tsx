import React from "react";
import { Controller, ControllerProps, FieldValues } from "react-hook-form";

type NativeInputs = 'input' | 'select' | 'textarea';

type Props = ControllerProps<React.ReactElement | React.ComponentType<any> | NativeInputs, FieldValues> & FormProps;

type FormProps = {
  name: string;
  className?: string;
  label?: string;
  type?: React.InputHTMLAttributes<unknown>['type'];
}

export function FormField({ defaultValue = '', ...props }: Props) {
  return (
    <Controller
      defaultValue={defaultValue}
      fullWidth={true} size="small" variant="outlined"
      {...props}
    />
  );

}