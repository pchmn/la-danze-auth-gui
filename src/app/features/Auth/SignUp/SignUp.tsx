import Button from '@material-ui/core/Button';
import LinearProgress from '@material-ui/core/LinearProgress';
import React from 'react';
import { AnimatedRoute, AnimatedSwitch } from '../../../shared/animation/RouteTransition';
import { FormikTextField } from '../../../shared/components/FormikTextField';
import { useSignUp } from './SignUp.hooks';
import styles from './SignUp.module.scss';


export function SignUp() {
  const { formik, loading, path, t } = useSignUp();

  return (
    <>
      <AnimatedSwitch>
        <AnimatedRoute exact path={path}>
          <h2 className={styles.title}>{t('auth.signUp.title')}</h2>
          <form className={styles.form} onSubmit={formik.handleSubmit}>
            <FormikTextField className={styles.textField} formik={formik} field="email" label={t('auth.signUp.email')} />
            <FormikTextField className={styles.textField} formik={formik} field="username" label={t('auth.signUp.username')} />
            <span className={styles.passwordHelper}>{t('auth.signUp.passwordHelper')}</span>
            <FormikTextField className={styles.textField} type="password" formik={formik} field="password" label={t('auth.password')} />
            <FormikTextField className={styles.textField} type="password" formik={formik} field="confirmPassword" label={t('auth.signUp.confirmPassword')} />
            <Button className={`${styles.submitButton} loadingButton`} disabled={!!loading} variant="contained" color="primary" type="submit">
              {t('common.form.validate')}
            </Button>
          </form>

          {loading &&
            <LinearProgress className={styles.progress} />
          }
        </AnimatedRoute>

        <AnimatedRoute exact path={`${path}/account-created`}>
          <div>
            Account created
            </div>
        </AnimatedRoute>
      </AnimatedSwitch>
    </>
  );
}
