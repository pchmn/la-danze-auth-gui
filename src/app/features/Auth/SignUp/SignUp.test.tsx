import { MockedProvider } from '@apollo/client/testing';
import { wait } from '@testing-library/react';
import { mount } from 'enzyme';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { FormField } from '../../../shared/components/FormField';
import { SignUp } from './SignUp';

const mocks: any[] = [];
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
}));

describe('<SignUp />', () => {
  const originalConsoleError = console.error;

  beforeEach(() => {
    console.error = jest.fn((msg) => {
      if (msg.includes('Warning: useLayoutEffect does nothing on the server')) {
        return null;
      } else {
        originalConsoleError(msg);
      }
    });
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  const container = mount(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter>
        <SignUp />
      </BrowserRouter>
    </MockedProvider>
  );

  const form = container.find('form');

  test('it should mount', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('it should show signUp form', () => {
    expect(container.find('h2').first().text()).toEqual('auth.signUp.title');
    expect(container.find('form').length).toEqual(1);
  });

  test('it should check mandatory fields', async () => {
    // Simulate submit
    form.simulate('submit');
    await wait();
    container.update();

    // Check errors
    // All fields mandatory
    container.find(FormField).forEach(wrapper => {
      expect(wrapper.props()).toEqual(
        expect.objectContaining({
          error: true,
          helperText: 'common.errors.requiredField'
        })
      );
    })
  });

  test('it should check email format', async () => {
    container.find('input[name="email"]').first().simulate('change', {
      target: { value: 'wrong@email.' }
    });
    // Simulate submit
    form.simulate('submit');
    await wait();
    container.update();

    // Check errors
    expect(container.find(FormField).first().props()).toEqual(
      expect.objectContaining({
        error: true,
        helperText: 'common.errors.invalidField'
      })
    );
  });

});
