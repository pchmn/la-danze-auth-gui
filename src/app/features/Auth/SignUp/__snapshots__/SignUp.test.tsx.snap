// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> it should mount 1`] = `"<span class=\\"signInHelper\\">auth.alreadyAnAccount<a href=\\"/signin\\">auth.signIn.title</a></span><div class=\\"MuiPaper-root AuthPaper MuiPaper-elevation1 MuiPaper-rounded\\"><h2 class=\\"title\\">auth.signUp.title</h2><form class=\\"form\\"><div class=\\"MuiFormControl-root MuiTextField-root textField MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined\\" data-shrink=\\"false\\">auth.signUp.email</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense\\"><input aria-invalid=\\"false\\" name=\\"email\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>auth.signUp.email</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root textField MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined\\" data-shrink=\\"false\\">auth.signUp.username</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense\\"><input aria-invalid=\\"false\\" name=\\"username\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>auth.signUp.username</span></legend></fieldset></div></div><span class=\\"passwordHelper\\">auth.signUp.passwordHelper</span><div class=\\"MuiFormControl-root MuiTextField-root textField MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined\\" data-shrink=\\"false\\">auth.password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense\\"><input aria-invalid=\\"false\\" name=\\"password\\" type=\\"password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>auth.password</span></legend></fieldset></div></div><div class=\\"MuiFormControl-root MuiTextField-root textField MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined\\" data-shrink=\\"false\\">auth.signUp.confirmPassword</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense\\"><input aria-invalid=\\"false\\" name=\\"confirmPassword\\" type=\\"password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>auth.signUp.confirmPassword</span></legend></fieldset></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained submitButton loadingButton MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">common.form.validate</span><span class=\\"MuiTouchRipple-root\\"></span></button></form></div>"`;
